lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -133,6 +163,7 @@ def test__has_access_descriptor(self):
         # TODO: override DISABLE_START_DATES and test the start date branch of the method
         user = Mock()
         descriptor = Mock(user_partitions=[])
+        descriptor._class_tags = {}
```

Comment:
It's needed in _has_access_descriptor, it checks if "detached" is in the class tags